pkg/storage/pebble.go

```diff
@@ -1038,15 +1039,35 @@ func (p *Pebble) Close() {
 		return
 	}
 	p.closed = true
-	_ = p.db.Close()
+
+	handleErr := func(err error) {
+		if err == nil {
+			return
+		}
+		// Allow unclean close in production builds for now. We refrain from
+		// Fatal-ing on an unclean close because Cockroach opens and closes
+		// ephemeral engines at time, and an error in those codepaths should not
+		// fatal the process.
+		//
+		// TODO(jackson): Propagate the error to call sites without fataling:
+		// This is tricky, because the Reader interface requires Close return
+		// nothing.
+		if buildutil.CrdbTestBuild {
+			p.logger.Fatalf("error during engine close: %s", err)
+		} else {
+			p.logger.Infof("error during engine close: %s", err)
```

Comment:
I'd promote this to an `Errorf`.

Reply:
the pebble Logger interface doesn't have an `Errorf`—updated to stash the logging context on the engine and use `log.Errorf`